src/libraries/System.Security.Cryptography.Cose/tests/CoseSign1MessageTests.Sign.cs

```diff
@@ -527,7 +515,35 @@ public void SignWithCborNegativeIntegerRepresentationAlgorithmHeaderValue(ulong
             CoseHeaderMap protectedHeaders = new CoseHeaderMap();
             protectedHeaders.SetEncodedValue(CoseHeaderLabel.Algorithm, encodedValue);
 
-            Assert.Throws<CryptographicException>(() => CoseSign1Message.Sign(s_sampleContent, DefaultKey, DefaultHash, protectedHeaders));
+            Assert.Throws<CryptographicException>(() => Sign(s_sampleContent, DefaultKey, DefaultHash, protectedHeaders));
+        }
+    }
+
+    public class CoseSign1Message_Sign_ByteArray : CoseSign1MessageTests_Sign
+    {
+        internal override byte[] Sign(byte[] content, AsymmetricAlgorithm key, HashAlgorithmName hashAlgorithm, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null, bool isDetached = false)
+            => CoseSign1Message.Sign(content, key, hashAlgorithm, protectedHeaders, unprotectedHeaders, isDetached);
+    }
+
+    public class CoseSign1Message_TrySign : CoseSign1MessageTests_Sign
+    {
+        internal override byte[] Sign(byte[] content!!, AsymmetricAlgorithm key!!, HashAlgorithmName hashAlgorithm, CoseHeaderMap? protectedHeaders = null, CoseHeaderMap? unprotectedHeaders = null, bool isDetached = false)
+        {
+            byte[] buffer = new byte[16]; // start small and double size until is big enough.
+
+            int bytesWritten;
+            while (!CoseSign1Message.TrySign(content, buffer, key, hashAlgorithm, out bytesWritten, protectedHeaders, unprotectedHeaders, isDetached))
+            {
+                Assert.Equal(0, bytesWritten);
+                Array.Resize(ref buffer, buffer.Length * 2);
+            }
+
+            ReadOnlySpan<byte> encodedMsg = buffer.AsSpan(0, bytesWritten);
+
+            int nonSpanOverloadLength = CoseSign1Message.Sign(content, key, hashAlgorithm, protectedHeaders, unprotectedHeaders, isDetached).Length;
+            Assert.Equal(nonSpanOverloadLength, bytesWritten);
```

Comment:
If we don't have "goldilocks" tests elsewhere, we could rewrite this helper:

* Call the array-returning one first.
* Call TrySign with a destination that's too small (mama bear?)
  * Can use the output array sliced small as the destination, and can do it many times, if desired.
* Allocate an array that's too big.  Call TrySign while writing to that (papa bear, I guess)
* Verify that bytesWritten is correct.
* Slice the too-big array down to bytesWritten, Clear() it for good measure, and call TrySign one more time (baby bear).

The problem with using that approach universally is that it's no good for testing exceptions, since those would have been handled by the call to the array-returning one.

Reply:
Done in https://github.com/dotnet/runtime/pull/66741/commits/e9888b5598ad159cf41d36d7250741520fda7ed3#diff-bbf49a8d3267b59f581d37edfe05375b94e73ee11ab6bf9e27870fe9a92dc633R441-R460
> The problem with using that approach universally is that it's no good for testing exceptions, since those would have been handled by the call to the array-returning one.

We can consider add another class `CoseSign1MessageTests_TrySign_Simple : CoseSign1MessageTests_Sign_CustomHeaderMaps` that only calls TrySign and doesn't do this bounds tests.